src/csharp/Grpc.Core/Logging/LogLevelFilterLogger.cs

```diff
@@ -27,16 +27,31 @@ namespace Grpc.Core.Logging
     /// <summary>Logger that filters out messages below certain log level.</summary>
     public class LogLevelFilterLogger : ILogger
     {
+        // Verbosity environment variable used by C core.
+        private const string CoreVerbosityEnvVarName = "GRPC_VERBOSITY";
         readonly ILogger innerLogger;
         readonly LogLevel logLevel;
 
         /// <summary>
         /// Creates and instance of <c>LogLevelFilter.</c>
         /// </summary>
-        public LogLevelFilterLogger(ILogger logger, LogLevel logLevel)
+        public LogLevelFilterLogger(ILogger logger, LogLevel logLevel) : this(logger, logLevel, false)
+        {
+        }
+
+        /// <summary>
+        /// Creates and instance of <c>LogLevelFilter.</c>
+        /// The <c>fromEnvironmentVariable</c> parameter allows looking up "GRPC_VERBOSITY" setting provided by C-core
+        /// and uses the same log level for C# logs. Using this setting is recommended as it makes the otherwise separate
```

Comment:
Comment nit: rather than lockstep, it might make sense to say here that the c core logging environment variable can override the c# log level, but not vice versa. IMO that could make use of the c# warning level more clear